include/dds/core/xtypes/detail/Annotation.hpp

```diff
@@ -29,34 +29,35 @@ namespace detail {
 class Annotation
 {
 public:
+    virtual ~Annotation() = default;
+
+    const AnnotationKind& kind() { return kind_; }
+
+protected:
     Annotation(
-            AnnotationKind ak)
-        : ak_(ak)
+            AnnotationKind kind)
+        : kind_(kind)
     {}
 
-    const AnnotationKind& akind() {return ak_;}
 private:
-    AnnotationKind ak_;
+    AnnotationKind kind_;
 };
 
-class IdAnnotation : public  Annotation
+class IdAnnotation : public Annotation
 {
 public:
-    IdAnnotation()
+    IdAnnotation(uint32_t id)
         : Annotation(AnnotationKind::OPTIONAL_ANNOTATION_TYPE)
-        , id_()
-    {} 
+        , id_(id)
+    {}
 
-    void id(
-            uint32_t id )
-    {
-        id_ = id;
-    }
+    IdAnnotation(const IdAnnotation& annotation)
+        : Annotation(AnnotationKind::ID_ANNOTATION_TYPE)
+        , id_(annotation.id())
+    {}
 
-    uint32_t id() const noexcept
-    {
-        return id_;
-    }
+    void id(uint32_t id) { id_ = id; }
+    uint32_t id() const { return id_; }
```

Comment:
we'll fight for the right of inline single line setters/getters

Reply:
:+1: